cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -976,6 +983,32 @@ public void getServiceInstances(HttpRequest request, HttpResponder responder,
   }
 
   /**
+   * Return the
+   */
+  @GET
+  @Path("/apps/{app-id}/services/{service-id}/active")
+  public void getServiceStatus(HttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId,
+                               @PathParam("app-id") String appId,
+                               @PathParam("service-id") String serviceId) throws NotFoundException {
+    ProgramId programId = Ids.namespace(namespaceId).app(appId).service(serviceId);
+    ProgramStatus status = lifecycleService.getProgramStatus(programId);
+    if (status == ProgramStatus.STOPPED) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Service is stopped. Please start it.");
```

Comment:
Also, if you stop and then restart a service, it will be "not running" (that is code will come here) for a short time, preceded and followed by time where it is running but shutting down or starting up. To an external caller, all three states should be the same, I think it would be more confusing if  the client gets 503 back, then retries and now gets a 400, after which it should - strictly speaking - not try again.